### PR TITLE
Us32 movement tweaking

### PR DESCRIPTION
**Pull request by Wessebanan**
Tweaked movement and fixed/changed some other stuff.

Try it out and let me know if it's shit.

### CODE REVIEW

File: Game/GameMain/gameAI/AISystems.cpp [hunk 1]
```diff
@@ -1261,9 +1263,14 @@ ecs::systems::RemoveDeadUnitsSystem::~RemoveDeadUnitsSystem()
 void ecs::systems::RemoveDeadUnitsSystem::updateEntity(FilteredEntity& entity, float delta)
 {
 	// The killers ID
-	unsigned int killer_id = getComponentFromKnownEntity<HealthComponent>(entity.entity->getID())->mHitBy;
-	// DEATH EFFECTS	
-	DeadComponent* p_dead = getComponentFromKnownEntity<DeadComponent>(entity.entity->getID());
+	unsigned int killer_id = 0;
+	HealthComponent* unit_health = entity.getComponent<HealthComponent>();
```
Comment (Heso113) [inline Game/GameMain/gameAI/AISystems.cpp]: U forgot the peepee! UwU

File: Game/GameMain/gameAI/AISystems.cpp [hunk 2]
```diff
@@ -1307,7 +1314,7 @@ void ecs::systems::RemoveDeadUnitsSystem::updateEntity(FilteredEntity& entity, f
 		}
 	}
 	//Fetch the units weapon data.
-	EquipmentComponent* equipment_comp = ECSUser::getComponentFromKnownEntity<EquipmentComponent>(entity.entity->getID());
+	EquipmentComponent* equipment_comp = entity.getComponent<EquipmentComponent>();
```
Comment (Heso113) [inline Game/GameMain/gameAI/AISystems.cpp]: U forgot the peepee! UwU

File: Game/GameMain/main.cpp [hunk 3]
```diff
@@ -241,16 +241,42 @@ void DebuggFunctions(EntityComponentSystem& rECS)
...
+	//	if (p_tile->tileType != WATER /*&& (count % ((rand() % 4) + 8)) == 0*/)
+	//	{
+	//		spawn_event.type = traps[rand() % (sizeof(traps) / sizeof(GAME_OBJECT_TYPES))];// GAME_OBJECT_TYPES((rand() % TRAP_TYPE_COUNT) + (GAME_OBJECT_TYPE_TRAP_OFFSET_TAG + 1));
+	//		spawn_event.tileID = p_tile->getEntityID();
+	//		rECS.createEvent(spawn_event);
+	//	}
+
+	//	count++;
+	//}
+
```
Comment (Heso113) [inline Game/GameMain/main.cpp]: Remove? UwU

Comment (Wessebanan) [reply Game/GameMain/main.cpp]: òmò used for debugging do not remove.

**Pull request closed without merging**